cli/test/e2e/package_authoring_e2e_test.go

```diff
@@ -163,89 +193,413 @@ spec:
+			require.Equal(t, expectedVendirOutput, out, "Expected vendir output to match")
+		})
 
-		// Verify Package artifact
-		out, err = readFile("./carvel-artifacts/packages/testpackage.corp.dev/package.yml")
-		require.NoErrorf(t, err, "Expected to read package.yml")
+		logger.Section(fmt.Sprintf("%s: Package release", testcase.Name), func() {
+			releaseInteraction := Interaction{
+				Prompts: []string{"Enter the registry URL"},
+				Inputs:  []string{env.Image},
+			}
 
-		expectedPackage = strings.TrimSpace(replaceSpaces(expectedPackage))
-		out = clearKeys(keysToBeIgnored, strings.TrimSpace(replaceSpaces(out)))
-		require.Equal(t, expectedPackage, out, "Expected Package to match")
-	})
+			go releaseInteraction.Run(promptOutput)
+
+			kappCtrl.RunWithOpts([]string{"pkg", "release", "--version", "1.0.0", "--tty=true", "--chdir", workingDir},
+				RunOpts{NoNamespace: true, StdinReader: promptOutput.StringReader(),
+					StdoutWriter: promptOutput.BufferedOutputWriter(), Interactive: true})
+
+			keysToBeIgnored := []string{"creationTimestamp:", "releasedAt:", "image"}
+
+			var pkgDir = "./carvel-artifacts/packages/testpackage.corp.dev/"
+
+			// Verify PackageMetadata artifact
+			out, err := readFile(pkgDir + "metadata.yml")
+			require.NoErrorf(t, err, "Expected to read metadata.yml")
+
+			expectedPackageMetadata := strings.TrimSpace(replaceSpaces(testcase.ExpectedPackageMetadata))
+			out = clearKeys(keysToBeIgnored, strings.TrimSpace(replaceSpaces(out)))
+			require.Equal(t, expectedPackageMetadata, out, "Expected PackageMetadata to match")
+
+			// Verify Package artifact
+			out, err = readFile(pkgDir + "package.yml")
+			require.NoErrorf(t, err, "Expected to read package.yml")
+
+			expectedPackage := strings.TrimSpace(replaceSpaces(testcase.ExpectedPackage))
+			out = clearKeys(keysToBeIgnored, strings.TrimSpace(replaceSpaces(out)))
+			require.Equal(t, expectedPackage, out, "Expected Package to match")
+		})
+
+		logger.Section(fmt.Sprintf("%s: Testing and installing created Package", testcase.Name), func() {
+
+			cleanUpInstalledPkg := func() {
+				switch testcase.Name {
+				case "Github Release Flow":
+					kubectl.RunWithOpts([]string{"delete", "ns", "dynatrace"}, RunOpts{NoNamespace: true})
+				}
+				kappCli.RunWithOpts([]string{"delete", "-a", "test-package"},
+					RunOpts{StdinReader: promptOutput.StringReader(), StdoutWriter: promptOutput.BufferedOutputWriter()})
+				kappCtrl.RunWithOpts([]string{"pkg", "installed", "delete", "-i", "test"},
+					RunOpts{StdinReader: promptOutput.StringReader(), StdoutWriter: promptOutput.BufferedOutputWriter()})
+			}
+
+			defer cleanUpInstalledPkg()
+
+			switch testcase.Name {
+			case "Github Release Flow":
+				kubectl.RunWithOpts([]string{"create", "ns", "dynatrace"}, RunOpts{NoNamespace: true})
+			}
+
+			kappCli.RunWithOpts([]string{"deploy", "-a", "test-package", "-f", fmt.Sprintf("%s/carvel-artifacts/packages/testpackage.corp.dev", workingDir), "-c"},
```

Comment:
This should use `pkgDir` as well. It will be resolved in #852 as it is adding tests for that particular case